File: app/src/main/java/com/cyl/musiclake/db/Album.java
package com.cyl.musiclake.db;

import com.cyl.musiclake.data.db.Music;

import org.litepal.crud.LitePalSupport;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by yonglong on 2016/11/23.
 */

public class Album extends LitePalSupport {

    private String id;
    private String name;
    private String artistName;
    private String cover;
    private long artistId;
    private int count;

    private List<Music> songs = new ArrayList<>();

    public Album() {
    }

    public Album(String id, String name, String artistName, long artistId, int count) {
        this.name = name;
        this.id = id;
        this.artistName = artistName;
        this.artistId = artistId;
        this.count = count;
    }

    public String getName() {
        return name;
    }

    public String getId() {
        return id;
    }

    public String getArtistName() {
        return artistName;
    }

    public long getArtistId() {
        return artistId;
    }

    public int getCount() {
        return count;
    }

    public String getCover() {
        return cover;
    }

    public void setCover(String cover) {
        this.cover = cover;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setArtistName(String artistName) {
        this.artistName = artistName;
    }

    public void setArtistId(long artistId) {
        this.artistId = artistId;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public List<Music> getSongs() {
        return songs;
    }

    public void setSongs(List<Music> songs) {
        this.songs = songs;
    }

    @Override
    public String toString() {
        return "Album{" +
                "name='" + name + '\'' +
                ", id=" + id +
                ", artistName='" + artistName + '\'' +
                ", artistId=" + artistId +
                ", count=" + count +
                '}';
    }
}

File: app/src/main/java/com/cyl/musiclake/db/Album.kt
package com.cyl.musiclake.db

import com.cyl.musiclake.common.Constants
import com.cyl.musiclake.data.db.Music
import org.litepal.crud.LitePalSupport
import java.io.Serializable
import java.util.*

/**
 * Created by yonglong on 2016/11/23.
 */

class Album : LitePalSupport, Serializable {

    var id: String? = null
    var name: String? = null
    var artistName: String? = null
    var cover: String? = null
    var type: String? = Constants.LOCAL
    var artistId: Long = 0
    var count: Int = 0

    var songs: List<Music> = ArrayList()

    constructor() {}

    constructor(id: String, name: String, artistName: String, artistId: Long, count: Int) {
        this.name = name
        this.id = id
        this.artistName = artistName
        this.artistId = artistId
        this.count = count
    }

    override fun toString(): String {
        return "Album{" +
                "name='" + name + '\''.toString() +
                ", id=" + id +
                ", artistName='" + artistName + '\''.toString() +
                ", artistId=" + artistId +
                ", count=" + count +
                '}'.toString()
    }
}
